src/UID2.Client/UID2Encryption.cs

```diff
@@ -56,6 +58,11 @@ internal static DecryptionResponse Decrypt(string token, KeyContainer keys, Date
 
         private static DecryptionResponse DecryptV2(byte[] encryptedId, KeyContainer keys, DateTime now, string domainName, ClientType clientType)
         {
+            if (encryptedId.Length != TOKEN_V2_LENGTH)
```

Comment:
A bit scary but as long as you're 100% certain it can't be 132 or 134 chars we should be ok :)